src/librustc/middle/resolve_lifetime.rs

```diff
@@ -463,28 +483,46 @@ impl<'a, 'tcx> Visitor<'tcx> for LifetimeContext<'a, 'tcx> {
     }
 
     fn visit_generics(&mut self, generics: &'tcx hir::Generics) {
-        for ty_param in generics.ty_params.iter() {
+        for ty_param in generics.ty_params() {
             walk_list!(self, visit_ty_param_bound, &ty_param.bounds);
             if let Some(ref ty) = ty_param.default {
                 self.visit_ty(&ty);
             }
         }
         for predicate in &generics.where_clause.predicates {
             match predicate {
-                &hir::WherePredicate::BoundPredicate(hir::WhereBoundPredicate{ ref bounded_ty,
-                                                                               ref bounds,
-                                                                               ref bound_lifetimes,
-                                                                               .. }) => {
-                    if !bound_lifetimes.is_empty() {
+                &hir::WherePredicate::BoundPredicate(hir::WhereBoundPredicate {
+                    ref bounded_ty,
+                    ref bounds,
+                    ref bound_generic_params,
+                    ..
+                }) => {
+                    if bound_generic_params.iter().find(|p| p.is_lifetime_param()).is_some() {
```

Comment:
This is just `.any(|p| p.is_lifetime_param())`, right?

Reply:
Yeah